pkg/keystore/repository.go

```diff
@@ -60,16 +57,35 @@ func (s *repository) Get(id string) (*Keystore, error) {
 }
 
 // Save stores the keystore.
-func (s *repository) Save(k *Keystore) error {
+func (r *repository) Save(k *Keystore) error {
+	if err := r.initStore(); err != nil {
+		return err
+	}
+
 	bytes, err := json.Marshal(k)
 	if err != nil {
 		return err
 	}
 
-	err = s.store.Put(k.ID, bytes)
+	err = r.store.Put(k.ID, bytes)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func (r *repository) initStore() error {
+	var err error
+
+	r.once.Do(func() {
```

Comment:
so `once.Do` will run only once in the lifetime of the server application being up.. the minute the server goes down and restarts, this will execute again.. are we sure we want to create the keystore again on server restart if the user client sends a CreateStore again?

Reply:
the store will not be created again - this is where `storage.ErrDuplicateStore` comes into the play - in your scenario store will be just "opened"